### ISSUE TITLE
Overflow in log loss function (use scipy.special.expit instead)

### ISSUE THREAD

**Issue by aldanor** [scikit-learn-contrib/lightning#121]
This often pops up when using log loss function:

```python
.../lightning/impl/base.py:55: RuntimeWarning: overflow encountered in exp
  prob = 1.0 / (1.0 + np.exp(-df))
```

which is to be expected; e.g., the following will cause this warning:

```python
>>> 1 / (1 + np.exp(1000))
```

Instead, [`scipy.special.expit`](https://docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.special.expit.html) can be used which is more robust and doesn't suffer from overflow problems.

**Comment by mblondel** [comment]
Pull-request welcome :)

**Comment by aldanor** [comment]
@mblondel Done in #122 - would it be possible to release a 0.5.1 + conda packages if/when this gets merged?

**Comment by mblondel** [comment]
Thanks for the fix, closing. 

@fabianp is taking care of releases but he might be busy lately.

**Comment by fabianp** [comment]
There are issues with the conda package: https://github.com/conda-forge/sklearn-contrib-lightning-feedstock/pull/5

Help is appreciated
